cantherm/statmech/kinetics.py

```diff
@@ -0,0 +1,71 @@
+class Reaction:
+    '''
+    Attributes:
+
+        reactants ():
+
+        ts ():
+
+        temp ():
+
+        reac_type ():
+
+        products ():
+
+        tunneling ():
+
+        scale ():
+
+        rates ():
+
+        activation_energy ():
+
+    '''
+
+    def __init__(self,
+                 reactants,
+                 ts,
+                 temp,
+                 reac_type='Unimol',
+                 products=[],
+                 tunneling=None,
+                 scale=0.99):
+        self.reactants = reactants
+        self.ts = ts
+        self.temp = temp     #YES
+        self.reac_type = reac_type     #YES
+        self.products = products
+        self.tunneling = tunneling
+        self.scale = scale
+        self.rates = None     #YES
+        self.tunneling_coeff = None     #YES
+        self.q_ratio = None     #YES
+        return
+
+    ############################################################################
+
+    def calc_TST_rates(self):
+        '''
+            Calculates the transition state theory rate constants for the
+            reaction at the given temperatures.
+        '''
+        self.rates = [0] * len(self.temp)
+        self.tunneling_coeff = [0] * len(self.temp)
+        self.q_ratio = [0] * len(self.temp)
+
+        for i in range(len(self.temp)):
+            t = self.temp[i]
+            if self.reac_type == 'Unimol':
+                Q_react = self.reactants.calculate_Q(t)
+                Q_TS = self.ts.calculate_Q(t)
```

Comment:
So write now `cantherm/chemistry/molecule.py` doesn't have a function `calculate_Q(T)`, but we have all the pieces ready in `cantherm/statmech/partition_function.py` so it's just a matter of putting them together as an instance function for the `CMol` class.

Reply:
I think I've applied this, but I'm not sure if it works or not. I have notes from a conversation we had about how to put together tests, but they're in the office. I'm planning to try to grab them over the weekend

Reply:
[Here's](https://docs.pytest.org/en/latest/getting-started.html#create-your-first-test) an example from the pytest documentation which is a good simple example of unittests.

Reply:
For what to test the kinetics against see the output file from `examples/h-tranf/input2` 

https://github.com/jamesETsmith/CANTHERM/blob/c7eb8dab31fa0175a44c4428b716a3f6e540a44d/examples/h-transf/output#L338-L363

Reply:
We can talk more about this over Zoom next week if you want